storage/innobase/lock/lock0lock.cc

```diff
@@ -582,6 +593,80 @@ ATTRIBUTE_NOINLINE static bool wsrep_is_BF_lock_timeout(const trx_t &trx)
              << " query: " << wsrep_thd_query(trx.mysql_thd);
   return true;
 }
+
+/** Checks if a lock request for a new lock has to wait for request
+    lock2 in Galera.
+@param trx                  trx of new lock
+@param type_mode            precise mode of the new lock
+                            to set: LOCK_S or LOCK_X, possibly
+                            ORed to LOCK_GAP or LOCK_REC_NOT_GAP,
+                            LOCK_INSERT_INTENTION.
+@param lock2                another record lock; NOTE that
+                            it is assumed that this has a lock bit
+                            set on the same record as in the new
+	                    lock we are setting.
+@return TRUE if new lock has to wait for lock2 to be removed */
+
+ATTRIBUTE_NOINLINE ATTRIBUTE_COLD
+bool lock_rec_has_to_wait_wsrep(const trx_t* trx,
+                                const unsigned type_mode,
+                                const lock_t* lock2)
+{
+  const trx_t* trx2= lock2->trx;
+
+  if (trx->is_wsrep_UK_scan() &&
+      wsrep_thd_is_BF(trx2->mysql_thd, false))
+  {
```

Comment:
Please omit `{` when the body consists of a single statement.

Reply:
I keep this because it is easier to read as it is (long comment).